File: app/src/main/java/xyz/satoshicano/halalfoodchecker/fragment/HomeFragment.java
package xyz.satoshicano.halalfoodchecker.fragment;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import xyz.satoshicano.halalfoodchecker.R;
import xyz.satoshicano.halalfoodchecker.activity.FindHalalCameraActivity;

public class HomeFragment extends Fragment implements View.OnClickListener{
    @Override
    public void onClick(View v) {
        startActivity(new Intent(getContext(),FindHalalCameraActivity.class));

    }
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState){
        View v = inflater.inflate(R.layout.fragment_home, container, false);
        FloatingActionButton fab = v.findViewById(R.id.fab);
        fab.setOnClickListener(this);
        return v;
    }

    public HomeFragment() {}

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
    }

    @Override
    public void onDetach() {
        super.onDetach();
    }

}

File: app/src/main/java/xyz/satoshicano/halalfoodchecker/fragment/HomeFragment.kt
package xyz.satoshicano.halalfoodchecker.fragment

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.design.widget.FloatingActionButton
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import xyz.satoshicano.halalfoodchecker.R
import xyz.satoshicano.halalfoodchecker.activity.FindHalalCameraActivity

class HomeFragment : Fragment(), View.OnClickListener {
    override fun onClick(v: View) {
        startActivity(Intent(context, FindHalalCameraActivity::class.java))

    }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val v = inflater!!.inflate(R.layout.fragment_home, container, false)
        val fab = v.findViewById<FloatingActionButton>(R.id.fab)
        fab.setOnClickListener(this)
        return v
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onAttach(context: Context?) {
        super.onAttach(context)
    }

    override fun onDetach() {
        super.onDetach()
    }

}
